src/test/scala/edu/ie3/simona/model/participant/HpModelSpec.scala

```diff
@@ -231,55 +231,186 @@ class HpModelSpec
       }
     }
 
-    "determining the flexibility options" when {
-      "the house is heated up and storage has space" should {
-        "deliver positive flexibility" in {
-          val house = thermalHouse(18, 22)
-            .copy(ethLosses = WattsPerKelvin(200))
-          val grid = thermalGrid(house, Some(thermalStorage))
-          val hp = hpModel(grid)
-          // Tick, at which the house is heated up
-          val relevantData = hpData.copy(currentTick = 2763L)
-          val thermalState = ThermalGridState(
-            Some(
-              ThermalHouseState(
-                0L,
-                Celsius(21),
-                Kilowatts(80),
-              )
+    "determining the flexibility options for different states" should {
+      "deliver correct flexibility options" in {
+        val testCases
+            : TableFor3[ThermalGridState, HpState, (Double, Double, Double)] =
+          Table(
+            ("thermalState", "lastState", "expectedValues"),
+            // House is below lower temperature boundary
+            (
+              ThermalGridState(
+                Some(ThermalHouseState(0L, Celsius(15), Kilowatts(0))),
+                Some(ThermalStorageState(0L, KilowattHours(20), Kilowatts(0))),
+              ),
+              HpState(
+                isRunning = false,
+                0,
+                Some(hpData.ambientTemperature),
+                Kilowatts(0.0),
```

Comment:
Why is the expression changed from the double value?

Reply:
If you refer to the case lefthand where the lastHouseState was 21°C and 80 Kilowatts feedin, this one should still be in the table (labeled `Storage and house have remaining capacity`)